src/thor/triplegbuilder.cc

```diff
@@ -1180,14 +1180,16 @@ TripLegBuilder::Build(const AttributesController& controller,
     // Set the endnode of this directed edge as the startnode of the next edge.
     startnode = directededge->endnode();
 
-    // Save the opposing edge as the previous DirectedEdge (for name consistency)
-    const GraphTile* t2 =
-        directededge->leaves_tile() ? graphreader.GetGraphTile(directededge->endnode()) : graphtile;
-    if (t2 == nullptr) {
-      continue;
+    if (!directededge->IsTransitLine()) {
+      // Save the opposing edge as the previous DirectedEdge (for name consistency)
+      const GraphTile* t2 =
+          directededge->leaves_tile() ? graphreader.GetGraphTile(directededge->endnode()) : graphtile;
+      if (t2 == nullptr) {
+        continue;
+      }
+      GraphId oppedge = t2->GetOpposingEdgeId(directededge);
```

Comment:
@slvlirnoff we can make this a much smaller diff, instead of wrapping all this with an `if`. right after you try to get the opposing edge id, you should be able to check that `oppedge.Is_Valid()` is `true` before using it.